### PR TITLE
Benchmarking rules

### PR DESCRIPTION
**Pull request by klemense1**
mostly new labels for ltl evaluator

### CODE REVIEW

File: bark/world/evaluation/ltl/label_functions/ego_below_speed_limit_label_function.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#include "bark/world/observed_world.hpp"
+
+using bark::models::dynamic::StateDefinition;
+
+namespace bark {
+namespace world {
+namespace evaluation {
+
+EgoBelowSpeedLimitLabelFunction::EgoBelowSpeedLimitLabelFunction(
```
Comment (patrickhart) [inline bark/world/evaluation/ltl/label_functions/ego_below_speed_limit_label_function.cpp:19]: Why use BelowSpeedLimitLabelFunction and EgoBelowSpeedLimitLabelFunction? would this not be possible in one class?

File: bark/world/evaluation/ltl/label_functions/ego_leftmost_lane_label_function.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+LabelMap EgoLeftmostLaneLabelFunction::Evaluate(
+    const world::ObservedWorld& observed_world) const {
+  const auto& ego_agent = observed_world.GetEgoAgent();
+  const auto& ego_pos = observed_world.CurrentEgoPosition();
+
+  const auto right_lc = observed_world.GetRoadCorridor()
+                            ->GetLeftRightLaneCorridor(ego_pos)
+                            .first;
+  bool is_leftmost_lane;
+  if (right_lc) {
```
Comment (patrickhart) [inline bark/world/evaluation/ltl/label_functions/ego_leftmost_lane_label_function.cpp:30]: why is the right lane corridor used for the leftmost label function?

File: bark/world/evaluation/ltl/label_functions/ego_rightmost_lane_label_function.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+LabelMap EgoRightmostLaneLabelFunction::Evaluate(
+    const world::ObservedWorld& observed_world) const {
+  const auto& ego_agent = observed_world.GetEgoAgent();
+  const auto& ego_pos = observed_world.CurrentEgoPosition();
+
+  const auto right_lc = observed_world.GetRoadCorridor()
+                            ->GetLeftRightLaneCorridor(ego_pos)
+                            .second;
+  bool is_rightmost_lane;
+  if (right_lc) {
```
Comment (patrickhart) [inline bark/world/evaluation/ltl/label_functions/ego_rightmost_lane_label_function.cpp:30]: seems like a copy and paste issue, would this also not be possible in a single evaluator/label function?

File: bark/world/evaluation/ltl/label_functions/rightmost_lane_label_function.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+bool RightmostLaneLabelFunction::EvaluateAgent(
+    const world::ObservedWorld& observed_world,
+    const AgentPtr& other_agent) const {
+  const auto& pos = other_agent->GetCurrentPosition();
+
+  const auto right_lc =
+      other_agent->GetRoadCorridor()->GetLeftRightLaneCorridor(pos).second;
+  bool is_rightmost_lane;
+  if (right_lc) {
```
Comment (patrickhart) [inline bark/world/evaluation/ltl/label_functions/rightmost_lane_label_function.cpp:29]: this seems like a lot of code duplication. I do not fully grasp the label infrastructure, but it seems to take up a lot of build time. Maybe we should quantify this in the future?

**Review by patrickhart — APPROVED** [review]

File: bark/world/evaluation/ltl/label_functions/ego_rightmost_lane_label_function.cpp (hunk 3 above)
Comment (klemense1) [reply bark/world/evaluation/ltl/label_functions/ego_rightmost_lane_label_function.cpp:30]: you are right, there is a naming issue

**Pull request merged**